components/button/button-subtle.js

```diff
@@ -148,19 +146,20 @@ class ButtonSubtle extends ButtonMixin(RtlMixin(LitElement)) {
 
 				button:hover:not([disabled]) d2l-icon.d2l-button-subtle-icon,
 				button:focus:not([disabled]) d2l-icon.d2l-button-subtle-icon,
-				:host([active]:not([disabled])) button d2l-icon.d2l-button-subtle-icon {
+				:host([active]:not([disabled])) button d2l-icon.d2l-button-subtle-icon,
+				button:hover:not([disabled]) slot[name="icon"]::slotted(d2l-icon-custom),
+				button:focus:not([disabled]) slot[name="icon"]::slotted(d2l-icon-custom),
+				:host([active]:not([disabled])) slot[name="icon"]::slotted(d2l-icon-custom) {
 					color: var(--d2l-color-celestine-minus-1);
 				}
 
-				:host([icon]) d2l-icon.d2l-button-subtle-icon {
+				.d2l-button-subtle-has-icon d2l-icon.d2l-button-subtle-icon,
+				slot[name="icon"]::slotted(d2l-icon-custom) {
 					display: inline-block;
 				}
-				:host([icon][icon-right]) d2l-icon.d2l-button-subtle-icon {
-					right: var(--d2l-button-subtle-padding-inline-end);
-				}
-				:host([dir="rtl"][icon][icon-right]) d2l-icon.d2l-button-subtle-icon {
-					left: var(--d2l-button-subtle-padding-inline-end);
-					right: auto;
+				:host([icon-right]) .d2l-button-subtle-has-icon d2l-icon.d2l-button-subtle-icon,
+				:host([icon-right]) slot[name="icon"]::slotted(d2l-icon-custom) {
+					inset-inline: auto var(--d2l-button-subtle-padding-inline-end);
```

Comment:
Could this just be this or is the `auto` needed?

```suggestion
					inset-inline-end: var(--d2l-button-subtle-padding-inline-end);
```

Reply:
Seems like it does just work fine without it - will remove